### PR TITLE
[MAJOR WIP DNM] A Very Insane Stun Overhaul

### PR DESCRIPTION
**Pull request by silicons**
**Tentative changes, may or may not be coded into the branch right now:**
- [ ] Stamina is no longer linked to health. You still get stunned at 100 stamina but it does not consider health anymore. If you take stamina damage your click delay, or how fast you can act in game with most weapons, will take longer to cool down. This means no matter how low health you are it takes 4 disabler shots to stun you, if you somehow get a disabler. **update: nevermind, i'm not going to entirely uncouple it, but it is likely not going to be completely dependent on health like before. so yes, you'll still have a significant disadvantage on the stamina end at low health.**
- [ ] Stamina will no longer slow you down more than 1 decisecond/move, down from 4. It will not start slowing you down until it's over 33.
- [ ] Tasers no longer one-hit stun. They do 30 stamina damage, and stun if your stamina damage is over 55, or your stamina damage and your missing health below 50 is over 55 (so stuns on one hit if you're at 25 health or below). They will also briefly make the victim's clickdelay far higher. They also cause 2 seconds of confusion on hit. Their stuns will also be scaled by health/stamina damage.
- [ ] Taser stun mode has 50 tile range again, and has 12 shots to compensate for not one-hitting.
- [ ] Taser disable mode replaced with light laser mode, 10 laser damage/shot, max 10 shots.
- [ ] Ebows will no longer one-hit stun. Rather, they now do radiation damage, can hold up to 3(?) shots, and do 20 toxins damage instead. **WOOPS FORGOT TO STATE THIS THEY'LL STILL STUN BUT NOT ONE HIT**
- [ ] Neurotoxin/beepsky smash no longer stun immediately, beepsky smash randomly makes you drop items on every ticks, and neurotoxin will fuck up your stamina and [other effects pending] like tirizine but not stun. 
- [ ] Any other one-hit ranged stun weapons that aren't outright adminspawn will be addressed accordingly. I'm looking at you, dragnet. And not really a stun weapon, but why don't we hit temperature guns too so they don't become the new meta. 
- [ ] Most of the above and more will receive new scaling.

# This is not final.
**TL;DR: I want combat to have more than "click them before they click you with a taser".
I want combat stuns to be relevant but not THE method of combat.
I want stamina to be a useful combat mechanic without being an immediate game over due to the slowdown it adds that makes you basically helpless.**
Why: I think it'd be neat to try an overhaul to our generally stun-centric combat.
Ideas, thoughts, comments, suggestions accepted and appreciated.

**Once again, most of this is not perfectly coded in, comment on the ideas, not the code!**

Also, please give thoughts about limb-specific effects for damage, assuming you don't immediately see the word taser and nerf and scream close PR.

### CODE REVIEW

**Comment by Armhulen** [comment]
>Revolver damage dropped to 55, and has less effect on armored targets.

closed pr

**Comment by Armhulen** [comment]
>Taser disable mode replaced with light laser mode, 10 laser damage/shot, max 10 shots.

closed pr, x2

**Comment by kevinz000** [comment]
I never saw the hate for light lasers, I'm sure security can be responsible.

**Comment by kevinz000** [comment]
Razharas has suggested limb-specific effects for disablers and tasers, anyone have thoughts?

**Comment by Armhulen** [comment]
un-nerf revolvers is my thought

**Comment by Xhuis** [comment]
This seems like a stun overhaul PR with a bunch of other really questionable balance changes thrown in. Light laser mode sounds awful, as it makes lethal ranged weapons very common and accessible to security as well as anyone with an emag, since they can take one from the arrivals checkpoint (I think!) The revolver and ebow nerfs seem shoehorned, as do the reagent changes which could likely be implemented if the initial PR was received well.

In my eyes, if the reagent changes, ebow and revolver changes, and light laser were removed, I'd be willing to try it out. Disabler mode still has a purpose for shooting through windows, after all.

**Comment by Marflowerose** [comment]
Something like disorient or stamina damage to ebow, otherwise it's only usefulness relies on the target vomiting after emptying entire clip

**Comment by kevinz000** [comment]
@Marflowerose it'd have the same balance as tasers basically with added tox
@Xhuis that's mainly because those weapons will become extremely powerful if tasers lose one-hit as they'll still one-hit stun. the main reason for it is because I don't see a way to do this by nerfing one thing at a time and then rebalancing everything over time to match. Also, the only source of extra tasers is armory or you'd see me with an extra every round.

**Comment by Armhulen** [comment]
make the ebow inject 20u of skewium/rotatium

**Comment by kevinz000** [comment]
not removing ebow stuns i forgot to state it'll be like the taser, somewhat.

**Comment by kevinz000** [comment]
Revolver damage nerf dropped due to popular (see: discord yelling) demand.

**Comment by Xhuis** [comment]
I still struggle to see the justification behind light lasers. Giving security officers access to lethal weapons without having to enter the armory has massively backfired every time.

**Comment by KomradeSpectre** [comment]
Once this is more final, I'd be down for seeing how well this actually works ingame.

**Comment by InsaneHyena** [comment]
I approve of these changed, but NOT the revolver balance change. What the fuck were you thinking?

**Comment by Incomptinence** [comment]
Stamina being linked to health was the one good thing about it. 
In fact it didn't go far enough. 
Assistants should have a higher chance to push you down so they can spit on your weak ravaged body.
Punches should have a better knock out chance against the infirm same with everything else. 

I despise this weak coddling of the wounded just like mobile crit disgustingly enabling when we should be encouraging people the get bloody patched up at every turn. 

Also love how your replace it with snowflake taser specific health thresh holds like that makes so much more sense.

**Comment by Incomptinence** [comment]
Also forget revolver balance. 

Nuke ops in particular is going to be a gigglefest utterly easy for the ops basically shredding people into meat puddles unopposed incredible savings on Eshields too because non researched energy weapons will be laughably impotent against them on it's own without the threat of decent spam or instastun weapons.

**Comment by praisenarsie** [comment]
guys it's just an i ded hidden in this stun overhaul

**Comment by kevinz000** [comment]
Hmm, I'll make being wounded more of a disadvantage soon.

**Comment by kevinz000** [comment]
on light lasers: i'd like to hear kor's opinion of that (and the entire thing) before I continue, so I guess this is on hold for 16 hours or until he wakes up.

**Comment by kevinz000** [comment]
Revolver damage nerf has been axed but the general idea behind it when I planned this entire thing was
Now: 1 hit taser stun, 2 hit revolver kill
Without nerf: 2 hit taser stun, 2 hit revolver kill
With nerf: 2 hit taser stun, 3 hit revolver kill but only if you're armored in the hit areas

**Comment by ShiggyDiggyDo** [comment]
For PRs that are this game changing we should have server votes like /vg/
http://ss13.moe/index.php/poll

**Comment by praisenarsie** [comment]
i agree with @ShiggyDiggyDo we have polls for a reason

**Comment by CPTANT** [comment]
But we can have a server poll right?

Let's just debate this a bit more to finalize some details, then do a test merge that isn't too short and put up a poll.

**Comment by calibraptor** [comment]
>Stamina is no longer linked to health. 

I'd really rather not. It just makes sense that somebody already weak from their injuries would go down quickly.

>If you take stamina damage your click delay, or how fast you can act in game with most weapons, will take longer to cool down.

Your wording is kinda confusing here but I'm assuming this means that having lots of stamina damage means you attack slower. I actually like this idea.

> Stamina will no longer slow you down more than 1 decisecond/move, down from 4. It will not start slowing you down until it's over 33.

So, stamina damage slows people less now? Ehhhh...

>Tasers no longer one-hit stun.

I'd be in favor of experimenting with something like this, so long as melee stuns are still a one-shot.

> Taser stun mode has 50 tile range again, and has 12 shots to compensate for not one-hitting.

Disagree with the absolutely retarded amount of range, agree with the increased number of shots if it's made a two shot stun.

>Taser disable mode replaced with light laser mode, 10 laser damage/shot, max 10 shots.

WHAT. NO.

>Neurotoxin/beepsky smash no longer stun immediately, beepsky smash randomly makes you drop items on every ticks, and neurotoxin will fuck up your stamina and [other effects pending] like tirizine but not stun.

No thanks.

>Any other one-hit stun weapons that aren't outright adminspawn will be addressed accordingly. I'm looking at you, dragnet. And not really a stun weapon, but why don't we hit temperature guns too so they don't become the new meta.

Leave melee stuns alone.

**Comment by kevinz000** [comment]
i fully support a server poll but only after i finalize details and get testmerges.
melee stuns will be left alone in this pr.

**Comment by kevinz000** [comment]
@calibraptor stamina being linked to health 100% is bad because that means slowdowns are op. stamina slowdowns are being weakened for a reason. i will be making low health have a significant impact though.

**Comment by kevinz000** [comment]
also linking to health and capping at 100 health damage + stam damage with the new attack speed penalties means the lower health you are you actually attack faster if people try to stam damage you but that's also offset by the stuns so *shrug*
still on hold until i hear from kor on what he feels about the general idea.

**Comment by arsserpentarium** [comment]
Maybe you should do it more partially?
Just make stamina overhaul. Without nerfing stun methods. Also,look at polaris system with halloss. I think,it can be useful.
Also sihnificant part of IRL martial combat is based on stun. Have you ever hitted someone with chair to the face? If you won't kill him,you'll stun  and deal significant brute.

**Comment by ACCount12** [comment]
* Stun balance memes
* Weapon balance memes
* Security balance memes
* Chem balance memes
* Antag balance memes
👎 x 5

**Comment by arsserpentarium** [comment]
Try to introduce new mechanics. Then try to rebalance with it step by step.
GIGANTIC REBALANCE is almost always shit.

**Comment by arsserpentarium** [comment]
Just don't dare to make all this shit in one PR. Every single momen is very qurstionable.

**Comment by Magikarpet** [comment]
biggest concern is that this turns into #15671 with the tasers but realism is the most retarded argument you can make as you can survive being bombed, shot by an assault rifle, and survive the void of space if you drink cryo fluid and a bar drink

**Comment by Ausops** [comment]
honestly I'd be ok with a testmerge and poll later

**Comment by CPTANT** [comment]
@Magikarpet  The way stamina works is also addressed by this PR. There are some changes between then and now, especially if also the stamina slowdown is nerfed.

**Comment by XDTM** [comment]
Please keep some kind of fake damage system for hallucination's sake

**Comment by Tlaltecuhtli** [comment]
port pain

**Comment by WJohn** [comment]
I'm curious how light laser mode will go. While ranged lethal mode *is* powerful, they're also pretty damned weak so it's likely not a huge deal. Though this WILL be a pretty huge nerf to blobs and xenos who suffer a ton from lasers.

I appreciate revolvers being actually *possible* to survive two shots, that's nice. Though with the near removal of one-shot stuns, being abe to counter now that you can survive 2 shots isn't gonna mean much since you *yourself* are gonna need to land multiple hits in order to knock them down.

Still, anything to shake up the meta is great. Thanks for this, kevinz.

**Comment by Supermichael777** [comment]
Giving sec flashlights was unpopular, i dont know how the autism patrol will react to a real gun

**Comment by Bawhoppen** [comment]
This will make many powersgamers mad, but we don't want those types of people here anyways

**Comment by banthebantz** [comment]
Remove the light laser bullshittery, it is unrelated to the change and was denied ages ago.

**Comment by WJohn** [comment]
We can always test it for a little while and remove it if it doesn't work out, yeah?

**Comment by xTrainx** [comment]
the abosolute state of tg

**Comment by Arzion5** [comment]
I like that TG has realized just how shit the current one-hit stun system is, but this is a bad way to proceed with it.
>Taser stun deals 30 stamina damage but the stamina damage required to slow down is 33
>Getting rid of disabler shots (Thank fucking God, but that still leaves lasers in)
>Giving tasers extreme range from the lack of disablers and giving tasers 12 shots
>Giving the taser an in-built Imperial Flashlight function
>E-bows no longer stun on one hit and are basically the rest of the codebases' shitty interpretation of a decent item

**Comment by KorPhaeron** [comment]
>I like that TG has realized just how shit the current one-hit stun system is, but this is a bad way to proceed with it.

The story of every combat overhaul attempt ever

**Comment by KorPhaeron** [comment]
Should test merge this and see how it plays

**Comment by sawrge** [comment]
@kevinz000 you should maybe just sit down and shake this out with kor and not pay too much mind to the bikeshedding til there's a testmerge

**Comment by rustledjimm** [comment]
I agree with the testmerge. We can only hope for actual feedback rather than "this is different so I hate it ree".

**Comment by CPTANT** [comment]
I think slowdown on first taser hit is necessary. 

There is a balance somewhere between first shot making it impossible to dodge the second one and security not being able to catch anyone because their shots do nothing.

Either a move delay of +1 or +2 (if its +2, then for only a short while)

**Comment by CPTANT** [comment]
On stamina slowdown:

Current stamina slowdown are as follows:
0-24 stamina -> 0 
25-49 stamina -> +1
50-74 stamina -> +2
75-99 stamina -> +3
100 stamina would be +4 but it stuns anyway

What if we change this to:
0-32 stamina ->0
33-65 stamina ->+1
66-99 stamina -> +2

This can easily be coded by multiplying stamina damage by 0.67 in the slowdown code. I am still not a huge fan of decoupling real and stamina damge though.

**Comment by arsserpentarium** [comment]
The idea isn't bad. But just don't do it in one piece. You should introduce new mechanic with stamina. And overhaul stunning part by part.
Also don't disperse. You already have  several big refactors on your plate.

**Comment by arsserpentarium** [comment]
Also at first you should make techwebs not so fucking horrible.
You didn't finished one your big project and starts another. 
Are you planning to make something with similar quality again?

**Comment by kevinz000** [comment]
@CPTANT people complained it was a one hit kill and honestly slowdown is a killer in our combat, even if ranged stuns are nerfed by one.
@arsserpentarium in reality i work on things in parallel better than series sorry

**Comment by kevinz000** [comment]
also @arsserpentarium refactors dont' take as much thought as overhauls they're literally making the code more compatible and modular and better i don't need to put thought into it other than "make sure it isnt' bugged horribly before tagging it ready". the only other thing that is seriously a timesink is techwebs and the stun overhaul doesn't have to be done immediately i mostly opened the pr to gather opinion and slowly crank out work on it as this is going to require a lot of thought and debate and not just a rash pr which is why i even made a forum thread for it.

**Comment by CPTANT** [comment]
I think a move delay of +1 is still reasonable for being hit with a taser.

**Comment by arsserpentarium** [comment]
Then I'll tell you,what will happen. Testmerge is too short to find all problems of all questionable changes.
I think, that overhaul of stamina damage is good, because it not only opens way for stun rehaul, but also opens way for many new chemical drugs, that can be added.
Also I think, that there should be not only stamina, but also pain.
These changes can lead to stun overhaul, but they leads to much more possibilities.
When all with this system will be done, stun can be rehauled.But step by step.

**Comment by awesine** [comment]
CLOSE THIS FUCK YOU DONT EVEN BOTHER

le ebin meme

**Comment by Rinoka** [comment]
awesine you are quickly becoming a louder, less funny goof. and half the reason goof didn't get shit is because people saw him as funny. so...

anyway, for people saying to do it bit by bit: kevin pointed out, correctly, that if a taser no longer onehit stuns, revolvers & other lethal guns have a significant advantage over a taser compared to current build.

i think tasers, honestly, would be better off as dealing significant falloff stamina damage; to make them a bit more in line with things like the standard melee onehit stun system. making them better at short range would reduce the impact of them versus revolvers, which would hopefully be good. it'd only have issues against people who can significantly outpace the bullet, but... don't know how often that happens.

disabler shots should definitely at least cause hand drops, if nothing else - it'd be an underutilized but fairly Cool way for sec to disarm people from range. could work well for disabling legs as a non-stamina based slowdown, too. there should probably be things you can wear to defend against these, though.

overall the concept is good and i hope we see stun overhauls from this. ss13 right now is very stun-focused, and this is a step in the right direction, even if it's bigger than some are comfortable with.

**Comment by KieranCat** [comment]
Can we nerf chloral hydrate while we're at it with the memes, same for morphine because le chem stuns from traitor doctors reeeeeeee

**Comment by Rinoka** [comment]
morphine's not that amazingly powerful, though it could do with some accessibility tweaks.

chloral nerf would be good probably, or at least having the recipe be complicated a bit

**Comment by KieranCat** [comment]
@Rinoka Chloral's a decent stun that could go for a nerf because it's so easy and cheap to make. Any competent chemist can make a chloral-cyanide pill/syringe and at least weaken their target quickly.

**Comment by Pacmandevil** [comment]
>people not realizing that the only thing worse than insta stuns are 2 shot stuns that're impossible to recover from

**Comment by kevinz000** [comment]
@Pacmandevil how are the two shot stuns proposed impossible to recover from?

**Comment by kevinz000** [comment]
also when I continue I'll be adding limb-specific stamina damage.

**Comment by AdamElTablawy** [comment]
Chloral and morphine are utterly pathetic as weapons and syringe guns, barring the rapid syringe gun, are slow to reload. No nerf is required Rinoka.

@kevinz000 I think the worry is that with how long stamina damage takes to fade, you'll keep going down. Where you can have a buddy just shake you up if you get tazed, if you have all this nasty shit like confusion and stamina on you when you go down you stay down.

**Comment by CPTANT** [comment]
@AdamElTablawy  Being knocked down now increases your stamina regen, so not recovering after getting 100 stamina damage is no longer a thing.

**Comment by QualityVan** [comment]
Are stun batons being touched by this?

**Comment by kevinz000** [comment]
Melee stuns that aren't linked to the two bar chems will not be touched.
@AdamElTablawy this is better in that case of thing because it doesn't also make you 4x slower than normal.

**Comment by Pacmandevil** [comment]
@kevinz000 I made that statement assuming it'd be like disablers, where you're slowed down significantly even after getting up, making disablers pretty well a straight upgrade to the taser.

if it's not like this, and you move full speed after getting up, I think this'd be fine

**Comment by optimumtact** [comment]
cleared the low effort posts

**Comment by Kerbin-Fiber** [comment]
So uh yeah, the limb specific effects sound great, like having your right arm being shot at makes use use items slower for that arm(?), something to that effect I guess. sounds neat

**Comment by ACCount12** [comment]
More like "deleted every comment about this PR being shit", am I right?

**Comment by CPTANT** [comment]
limb specific damage effects are needed anyway.

We now have a system where limbs do have a damage cap, but no negative effects from actually reaching that cap.

I shot someone 12 times in the left arm (yes stuff like that happens) and while I can accept that he didn't go down, he then used the same arm to shoot me back, which is just stupid.

**Comment by outbools** [comment]
@ACCount12 there is no much "WOW THIS IS GREAT" comment either. Github comments for proposals and suggestions ONLY, if you not a member of /tg/ maintenance team your claims will be ignored until you find some arguments to support them.

**Comment by GuyonBroadway** [comment]
Suggestion for beepsky smash. Every other tick it stuns for about a second and plays a taser noise to the affected mob. With a chance each stun to play a beepsky line.

**Comment by KieranCat** [comment]
Like "Have a secure day"?

**Comment by AnturK** [comment]
Will reopen when you're done with other x staling refactors.
